src/platform/test.rs

```diff
@@ -173,41 +173,40 @@ fn big_data_with_sender_transfer() {
     thread.join().unwrap();
 }
 
-fn with_n_fds(n: usize, size: usize) {
-    let (sender_fds, receivers): (Vec<_>, Vec<_>) = (0..n).map(|_| platform::channel().unwrap())
-                                                    .map(|(tx, rx)| (OsIpcChannel::Sender(tx), rx))
-                                                    .unzip();
-    let (super_tx, super_rx) = platform::channel().unwrap();
-
-    let data: Vec<u8> = (0..size).map(|i| (i % 251) as u8).collect();
-    super_tx.send(&data[..], sender_fds, vec![]).unwrap();
-    let (mut received_data, received_channels, received_shared_memory_regions) =
-        super_rx.recv().unwrap();
+// These tests only apply to platforms that need fragmentation.
+#[cfg(all(not(feature = "inprocess"), target_os = "linux"))]
+mod fragment_tests {
+    use platform::{self, OsIpcChannel};
 
-    received_data.truncate(size);
-    assert_eq!(received_data.len(), data.len());
-    assert_eq!(&received_data[..], &data[..]);
-    assert_eq!(received_channels.len(), receivers.len());
-    assert_eq!(received_shared_memory_regions.len(), 0);
+    fn with_n_fds(n: usize, size: usize) {
```

Comment:
I actually left the function outside of the conditional block on purpose, as it's not really platform-specific, and some existing and/or newly added generic tests could start using it too... I don't feel strongly about this though.